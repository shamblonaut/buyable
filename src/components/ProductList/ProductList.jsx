import { useEffect, useState } from "react";
import { useSearchParams } from "react-router-dom";
import PropTypes from "prop-types";

import { ProductFilters, ProductCard } from "@/components";
import { Info } from "@/styles";
import { Container, List } from "./ProductList.styles";

const ProductList = ({ products, cart, setCart, filterable = true }) => {
  const [searchParams] = useSearchParams();
  const [filteredProducts, setFilteredProducts] = useState(products);

  useEffect(() => {
    if (!filterable) {
      setFilteredProducts(products);
    }
  }, [products, filterable]);

  const searchQuery = searchParams.get("query");
  return (
    <Container>
      {filterable && (
        <ProductFilters
          products={products}
          setFilteredProducts={setFilteredProducts}
        />
      )}
      <List>
        {filteredProducts.length > 0 ? (
          filteredProducts.map((product) => (
            <li key={product.id}>
              <ProductCard product={product} cart={cart} setCart={setCart} />
            </li>
          ))
        ) : (
          <Info>No products found{searchQuery && ` for "${searchQuery}"`}</Info>
        )}
      </List>
    </Container>
  );
};

ProductList.propTypes = {
  products: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired,
      price: PropTypes.number.isRequired,
      image: PropTypes.string.isRequired,
      category: PropTypes.string.isRequired,
      rating: PropTypes.exact({
        rate: PropTypes.number.isRequired,
        count: PropTypes.number.isRequired,
      }).isRequired,
    }).isRequired,
  ),
  cart: PropTypes.objectOf(PropTypes.number).isRequired,
  setCart: PropTypes.func.isRequired,
  filterable: PropTypes.bool,
};

export default ProductList;
