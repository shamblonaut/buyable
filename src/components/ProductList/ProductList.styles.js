import styled from "styled-components";

export const Container = styled.div`
  width: 100%;
  margin-bottom: 32px;

  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 16px;
`;

export const List = styled.ul`
  list-style: none;
  width: 100%;
  padding: 16px;
  display: grid;
  grid-template-columns: repeat(auto-fit, 360px);
  justify-content: center;
  gap: 16px;
`;
