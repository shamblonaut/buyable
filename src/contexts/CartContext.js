import { createContext } from "react";

const CartContext = createContext({
  cart: {},
  dispatchCartAction: () => {},
});

export default CartContext;
