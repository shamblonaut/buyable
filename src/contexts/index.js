import CartContext from "./CartContext";

export { CartContext };
