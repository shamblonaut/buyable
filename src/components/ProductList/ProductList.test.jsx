import { describe, it, expect, vi, beforeEach } from "vitest";
import { render, screen, within } from "@testing-library/react";

import { BrowserRouter } from "react-router-dom";

import { CartContext } from "@/contexts";

import { mockProductList } from "@tests/data";

import ProductList from "./ProductList";

const mockCart = { 1: 1, 3: 2, 5: 5, 7: 13, 9: 34 };
const mockDispatchCartAction = vi.fn();

vi.mock(
  "@/components/ProductCard/ProductCard.styles",
  async (importOriginal) => {
    const actual = await importOriginal();

    const { ProductLink } = actual;
    return {
      ...actual,
      ProductLink: ({ children, ...props }) => (
        <ProductLink data-testid="product-link" {...props}>
          {children}
        </ProductLink>
      ),
    };
  },
);

describe("ProductList component", () => {
  beforeEach(() => {
    vi.clearAllMocks();

    // Clear search queries
    history.replaceState(history.state, null, window.location.pathname);
  });

  it("renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <CartContext.Provider
          value={{ cart: mockCart, dispatchCartAction: mockDispatchCartAction }}
        >
          <ProductList products={mockProductList} />
        </CartContext.Provider>
      </BrowserRouter>,
    );

    expect(container).toMatchSnapshot();
  });

  it("does not show product filters when it is disabled", () => {
    const { rerender } = render(
      <BrowserRouter>
        <CartContext.Provider
          value={{ cart: mockCart, dispatchCartAction: mockDispatchCartAction }}
        >
          <ProductList products={mockProductList} />
        </CartContext.Provider>
      </BrowserRouter>,
    );

    expect(screen.queryByLabelText("Search for product")).toBeInTheDocument();

    rerender(
      <BrowserRouter>
        <CartContext.Provider
          value={{ cart: mockCart, dispatchCartAction: mockDispatchCartAction }}
        >
          <ProductList products={mockProductList} filterable={false} />
        </CartContext.Provider>
      </BrowserRouter>,
    );

    expect(
      screen.queryByLabelText("Search for product"),
    ).not.toBeInTheDocument();
  });

  it("shows message if no products are found", () => {
    render(
      <BrowserRouter>
        <CartContext.Provider
          value={{ cart: mockCart, dispatchCartAction: mockDispatchCartAction }}
        >
          <ProductList products={[]} />
        </CartContext.Provider>
      </BrowserRouter>,
    );

    expect(
      screen.getByText("no products found", { exact: false }),
    ).toBeInTheDocument();
  });

  it("shows appropriate message when no products are found for search query", () => {
    const searchQuery = "proust";
    history.replaceState(
      history.state,
      null,
      `${window.location.pathname}?query=${searchQuery}`,
    );

    render(
      <BrowserRouter>
        <CartContext.Provider
          value={{ cart: mockCart, dispatchCartAction: mockDispatchCartAction }}
        >
          <ProductList products={mockProductList} />
        </CartContext.Provider>
      </BrowserRouter>,
    );

    expect(
      screen.getByText(`no products found for "${searchQuery}"`, {
        exact: false,
      }),
    ).toBeInTheDocument();
  });

  it("shows all products in a list if not filtered", () => {
    render(
      <BrowserRouter>
        <CartContext.Provider
          value={{ cart: mockCart, dispatchCartAction: mockDispatchCartAction }}
        >
          <ProductList products={mockProductList} />
        </CartContext.Provider>
      </BrowserRouter>,
    );

    expect(
      within(screen.getByRole("list")).getAllByTestId("product-link").length,
    ).toBe(mockProductList.length);
  });

  it("shows only the filtered products if a search query is present", () => {
    const searchQuery = "Electronics";
    history.replaceState(
      history.state,
      null,
      `${window.location.pathname}?query=${searchQuery}`,
    );

    render(
      <BrowserRouter>
        <CartContext.Provider
          value={{ cart: mockCart, dispatchCartAction: mockDispatchCartAction }}
        >
          <ProductList products={mockProductList} />
        </CartContext.Provider>
      </BrowserRouter>,
    );

    expect(
      within(screen.getByRole("list")).getAllByTestId("product-link").length,
    ).toBe(5);
  });
});
