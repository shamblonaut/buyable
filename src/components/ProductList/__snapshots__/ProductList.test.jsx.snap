// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductList component > renders correctly 1`] = `
<div>
  <div
    class="sc-iIvHqT iRLXnh"
  >
    <div
      class="sc-jaXbil dCfsje"
    >
      <form
        class="sc-gLaqbQ cyMkYY"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-search sc-ipUnzB eqeLE"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m21 21-4.34-4.34"
          />
          <circle
            cx="11"
            cy="11"
            r="8"
          />
        </svg>
        <input
          aria-label="product-search-input"
          class="sc-eQwNpu leZVnm"
          placeholder="Search Product"
          type="text"
          value=""
        />
      </form>
      <div
        class="sc-eqNDNG fQPfRZ"
      >
        <div
          class="sc-fVHBlr dXkiUT"
        >
          <label
            class="sc-cOpnSz euyVPc"
            for="category-selector"
          >
            Category: 
          </label>
          <select
            aria-label="category-selector"
            class="sc-kjwdDK ka-DCaX"
            id="category-selector"
          >
            <option
              value="all"
            >
              All
            </option>
            <option
              value="Electronics"
            >
              Electronics
            </option>
            <option
              value="Accessories"
            >
              Accessories
            </option>
          </select>
        </div>
        <div
          class="sc-fVHBlr sc-gvqKNf dXkiUT bdBtbg"
        >
          <label
            class="sc-cOpnSz euyVPc"
            for="sort-selector"
          >
            Sort By: 
          </label>
          <select
            aria-label="sort-selector"
            class="sc-kjwdDK ka-DCaX"
            id="sort-selector"
          >
            <option
              value="relevance"
            >
              Relevance
            </option>
            <option
              value="cheap"
            >
              Price: Low to High
            </option>
            <option
              value="expensive"
            >
              Price: High to Low
            </option>
            <option
              value="best"
            >
              Best Rating
            </option>
          </select>
        </div>
      </div>
    </div>
    <ul
      class="sc-jxOwhs dcwtMO"
    >
      <li>
        <div
          class="sc-eVqvcJ OmFqL"
        >
          <a
            aria-label="product-link"
            class="sc-bbbBoY kaFTmX"
            data-discover="true"
            href="/product/1"
            title="Proton L20 32GB RAM 2TB SSD Laptop"
          >
            <div
              class="sc-cdmAjP kxojuH"
            >
              <img
                alt="Proton L20 32GB RAM 2TB SSD Laptop"
                src="http://example.com/image-1.jpg"
              />
            </div>
            <div
              class="sc-jytpVa gpNucy"
            >
              <p
                class="sc-eknHtZ jjmWJx"
              >
                Proton L20 32GB RAM 2TB SSD Laptop
              </p>
              <p
                class="sc-pYNGo jkBVsK"
              >
                $
                1599
              </p>
            </div>
          </a>
          <div
            class="sc-cCVJLD cFLicH"
          >
            <div
              aria-label="cart-quantity-selector"
              class="sc-etsjJW fLdKFf"
            >
              <button
                aria-label="quantity-decrement-button"
                class="sc-fYmhhH sc-koBvLg fcPUbH jQiuGK"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-minus"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                </svg>
              </button>
              <input
                aria-label="custom-quantity-input"
                class="sc-kUouGy hgimou"
                type="text"
                value="1"
              />
              <button
                aria-label="quantity-increment-button"
                class="sc-fYmhhH sc-DZJJV fcPUbH gbGTLy"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-plus"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                  <path
                    d="M12 5v14"
                  />
                </svg>
              </button>
            </div>
            <button
              aria-label="cart-remove-button"
              class="sc-jYczwO sc-fszimp bhIeZs fvTTVY"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-trash2 lucide-trash-2"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 6h18"
                />
                <path
                  d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                />
                <path
                  d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                />
                <line
                  x1="10"
                  x2="10"
                  y1="11"
                  y2="17"
                />
                <line
                  x1="14"
                  x2="14"
                  y1="11"
                  y2="17"
                />
              </svg>
            </button>
          </div>
        </div>
      </li>
      <li>
        <div
          class="sc-eVqvcJ OmFqL"
        >
          <a
            aria-label="product-link"
            class="sc-bbbBoY kaFTmX"
            data-discover="true"
            href="/product/2"
            title="Tungstenite G8 16GB VRAM Graphics Card"
          >
            <div
              class="sc-cdmAjP kxojuH"
            >
              <img
                alt="Tungstenite G8 16GB VRAM Graphics Card"
                src="http://example.com/image-2.jpg"
              />
            </div>
            <div
              class="sc-jytpVa gpNucy"
            >
              <p
                class="sc-eknHtZ jjmWJx"
              >
                Tungstenite G8 16GB VRAM Graphics Card
              </p>
              <p
                class="sc-pYNGo jkBVsK"
              >
                $
                699
              </p>
            </div>
          </a>
          <div
            class="sc-cCVJLD cFLicH"
          >
            <button
              aria-label="cart-add-button"
              class="sc-jYczwO sc-dNFkOE bhIeZs jFefEh"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-shopping-cart"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="8"
                  cy="21"
                  r="1"
                />
                <circle
                  cx="19"
                  cy="21"
                  r="1"
                />
                <path
                  d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
                />
              </svg>
               Add to cart
            </button>
          </div>
        </div>
      </li>
      <li>
        <div
          class="sc-eVqvcJ OmFqL"
        >
          <a
            aria-label="product-link"
            class="sc-bbbBoY kaFTmX"
            data-discover="true"
            href="/product/3"
            title="Element S12 8GB RAM 256GB ROM Smartphone"
          >
            <div
              class="sc-cdmAjP kxojuH"
            >
              <img
                alt="Element S12 8GB RAM 256GB ROM Smartphone"
                src="http://example.com/image-3.jpg"
              />
            </div>
            <div
              class="sc-jytpVa gpNucy"
            >
              <p
                class="sc-eknHtZ jjmWJx"
              >
                Element S12 8GB RAM 256GB ROM Smartphone
              </p>
              <p
                class="sc-pYNGo jkBVsK"
              >
                $
                249
              </p>
            </div>
          </a>
          <div
            class="sc-cCVJLD cFLicH"
          >
            <div
              aria-label="cart-quantity-selector"
              class="sc-etsjJW fLdKFf"
            >
              <button
                aria-label="quantity-decrement-button"
                class="sc-fYmhhH sc-koBvLg fcPUbH jQiuGK"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-minus"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                </svg>
              </button>
              <input
                aria-label="custom-quantity-input"
                class="sc-kUouGy hgimou"
                type="text"
                value="2"
              />
              <button
                aria-label="quantity-increment-button"
                class="sc-fYmhhH sc-DZJJV fcPUbH gbGTLy"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-plus"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                  <path
                    d="M12 5v14"
                  />
                </svg>
              </button>
            </div>
            <button
              aria-label="cart-remove-button"
              class="sc-jYczwO sc-fszimp bhIeZs fvTTVY"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-trash2 lucide-trash-2"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 6h18"
                />
                <path
                  d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                />
                <path
                  d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                />
                <line
                  x1="10"
                  x2="10"
                  y1="11"
                  y2="17"
                />
                <line
                  x1="14"
                  x2="14"
                  y1="11"
                  y2="17"
                />
              </svg>
            </button>
          </div>
        </div>
      </li>
      <li>
        <div
          class="sc-eVqvcJ OmFqL"
        >
          <a
            aria-label="product-link"
            class="sc-bbbBoY kaFTmX"
            data-discover="true"
            href="/product/4"
            title="Electron P11 16-Core Processor"
          >
            <div
              class="sc-cdmAjP kxojuH"
            >
              <img
                alt="Electron P11 16-Core Processor"
                src="http://example.com/image-4.jpg"
              />
            </div>
            <div
              class="sc-jytpVa gpNucy"
            >
              <p
                class="sc-eknHtZ jjmWJx"
              >
                Electron P11 16-Core Processor
              </p>
              <p
                class="sc-pYNGo jkBVsK"
              >
                $
                549
              </p>
            </div>
          </a>
          <div
            class="sc-cCVJLD cFLicH"
          >
            <button
              aria-label="cart-add-button"
              class="sc-jYczwO sc-dNFkOE bhIeZs jFefEh"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-shopping-cart"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="8"
                  cy="21"
                  r="1"
                />
                <circle
                  cx="19"
                  cy="21"
                  r="1"
                />
                <path
                  d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
                />
              </svg>
               Add to cart
            </button>
          </div>
        </div>
      </li>
      <li>
        <div
          class="sc-eVqvcJ OmFqL"
        >
          <a
            aria-label="product-link"
            class="sc-bbbBoY kaFTmX"
            data-discover="true"
            href="/product/5"
            title="Nucleon LGA 1700 Motherboard"
          >
            <div
              class="sc-cdmAjP kxojuH"
            >
              <img
                alt="Nucleon LGA 1700 Motherboard"
                src="http://example.com/image-5.jpg"
              />
            </div>
            <div
              class="sc-jytpVa gpNucy"
            >
              <p
                class="sc-eknHtZ jjmWJx"
              >
                Nucleon LGA 1700 Motherboard
              </p>
              <p
                class="sc-pYNGo jkBVsK"
              >
                $
                179
              </p>
            </div>
          </a>
          <div
            class="sc-cCVJLD cFLicH"
          >
            <div
              aria-label="cart-quantity-selector"
              class="sc-etsjJW fLdKFf"
            >
              <button
                aria-label="quantity-decrement-button"
                class="sc-fYmhhH sc-koBvLg fcPUbH jQiuGK"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-minus"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                </svg>
              </button>
              <input
                aria-label="custom-quantity-input"
                class="sc-kUouGy hgimou"
                type="text"
                value="5"
              />
              <button
                aria-label="quantity-increment-button"
                class="sc-fYmhhH sc-DZJJV fcPUbH gbGTLy"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-plus"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                  <path
                    d="M12 5v14"
                  />
                </svg>
              </button>
            </div>
            <button
              aria-label="cart-remove-button"
              class="sc-jYczwO sc-fszimp bhIeZs fvTTVY"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-trash2 lucide-trash-2"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 6h18"
                />
                <path
                  d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                />
                <path
                  d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                />
                <line
                  x1="10"
                  x2="10"
                  y1="11"
                  y2="17"
                />
                <line
                  x1="14"
                  x2="14"
                  y1="11"
                  y2="17"
                />
              </svg>
            </button>
          </div>
        </div>
      </li>
      <li>
        <div
          class="sc-eVqvcJ OmFqL"
        >
          <a
            aria-label="product-link"
            class="sc-bbbBoY kaFTmX"
            data-discover="true"
            href="/product/6"
            title="Newton Sports Sneakers"
          >
            <div
              class="sc-cdmAjP kxojuH"
            >
              <img
                alt="Newton Sports Sneakers"
                src="http://example.com/image-6.jpg"
              />
            </div>
            <div
              class="sc-jytpVa gpNucy"
            >
              <p
                class="sc-eknHtZ jjmWJx"
              >
                Newton Sports Sneakers
              </p>
              <p
                class="sc-pYNGo jkBVsK"
              >
                $
                49.99
              </p>
            </div>
          </a>
          <div
            class="sc-cCVJLD cFLicH"
          >
            <button
              aria-label="cart-add-button"
              class="sc-jYczwO sc-dNFkOE bhIeZs jFefEh"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-shopping-cart"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="8"
                  cy="21"
                  r="1"
                />
                <circle
                  cx="19"
                  cy="21"
                  r="1"
                />
                <path
                  d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
                />
              </svg>
               Add to cart
            </button>
          </div>
        </div>
      </li>
      <li>
        <div
          class="sc-eVqvcJ OmFqL"
        >
          <a
            aria-label="product-link"
            class="sc-bbbBoY kaFTmX"
            data-discover="true"
            href="/product/7"
            title="Graviton Travel Bag"
          >
            <div
              class="sc-cdmAjP kxojuH"
            >
              <img
                alt="Graviton Travel Bag"
                src="http://example.com/image-7.jpg"
              />
            </div>
            <div
              class="sc-jytpVa gpNucy"
            >
              <p
                class="sc-eknHtZ jjmWJx"
              >
                Graviton Travel Bag
              </p>
              <p
                class="sc-pYNGo jkBVsK"
              >
                $
                22.49
              </p>
            </div>
          </a>
          <div
            class="sc-cCVJLD cFLicH"
          >
            <div
              aria-label="cart-quantity-selector"
              class="sc-etsjJW fLdKFf"
            >
              <button
                aria-label="quantity-decrement-button"
                class="sc-fYmhhH sc-koBvLg fcPUbH jQiuGK"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-minus"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                </svg>
              </button>
              <input
                aria-label="custom-quantity-input"
                class="sc-kUouGy hgimou"
                type="text"
                value="13"
              />
              <button
                aria-label="quantity-increment-button"
                class="sc-fYmhhH sc-DZJJV fcPUbH gbGTLy"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-plus"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                  <path
                    d="M12 5v14"
                  />
                </svg>
              </button>
            </div>
            <button
              aria-label="cart-remove-button"
              class="sc-jYczwO sc-fszimp bhIeZs fvTTVY"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-trash2 lucide-trash-2"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 6h18"
                />
                <path
                  d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                />
                <path
                  d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                />
                <line
                  x1="10"
                  x2="10"
                  y1="11"
                  y2="17"
                />
                <line
                  x1="14"
                  x2="14"
                  y1="11"
                  y2="17"
                />
              </svg>
            </button>
          </div>
        </div>
      </li>
      <li>
        <div
          class="sc-eVqvcJ OmFqL"
        >
          <a
            aria-label="product-link"
            class="sc-bbbBoY kaFTmX"
            data-discover="true"
            href="/product/8"
            title="Torqus Chronograph Wristwatch"
          >
            <div
              class="sc-cdmAjP kxojuH"
            >
              <img
                alt="Torqus Chronograph Wristwatch"
                src="http://example.com/image-8.jpg"
              />
            </div>
            <div
              class="sc-jytpVa gpNucy"
            >
              <p
                class="sc-eknHtZ jjmWJx"
              >
                Torqus Chronograph Wristwatch
              </p>
              <p
                class="sc-pYNGo jkBVsK"
              >
                $
                79.99
              </p>
            </div>
          </a>
          <div
            class="sc-cCVJLD cFLicH"
          >
            <button
              aria-label="cart-add-button"
              class="sc-jYczwO sc-dNFkOE bhIeZs jFefEh"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-shopping-cart"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="8"
                  cy="21"
                  r="1"
                />
                <circle
                  cx="19"
                  cy="21"
                  r="1"
                />
                <path
                  d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
                />
              </svg>
               Add to cart
            </button>
          </div>
        </div>
      </li>
      <li>
        <div
          class="sc-eVqvcJ OmFqL"
        >
          <a
            aria-label="product-link"
            class="sc-bbbBoY kaFTmX"
            data-discover="true"
            href="/product/9"
            title="Pendulus Silver Necklace"
          >
            <div
              class="sc-cdmAjP kxojuH"
            >
              <img
                alt="Pendulus Silver Necklace"
                src="http://example.com/image-9.jpg"
              />
            </div>
            <div
              class="sc-jytpVa gpNucy"
            >
              <p
                class="sc-eknHtZ jjmWJx"
              >
                Pendulus Silver Necklace
              </p>
              <p
                class="sc-pYNGo jkBVsK"
              >
                $
                13.49
              </p>
            </div>
          </a>
          <div
            class="sc-cCVJLD cFLicH"
          >
            <div
              aria-label="cart-quantity-selector"
              class="sc-etsjJW fLdKFf"
            >
              <button
                aria-label="quantity-decrement-button"
                class="sc-fYmhhH sc-koBvLg fcPUbH jQiuGK"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-minus"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                </svg>
              </button>
              <input
                aria-label="custom-quantity-input"
                class="sc-kUouGy hgimou"
                type="text"
                value="34"
              />
              <button
                aria-label="quantity-increment-button"
                class="sc-fYmhhH sc-DZJJV fcPUbH gbGTLy"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-plus"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                  <path
                    d="M12 5v14"
                  />
                </svg>
              </button>
            </div>
            <button
              aria-label="cart-remove-button"
              class="sc-jYczwO sc-fszimp bhIeZs fvTTVY"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-trash2 lucide-trash-2"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 6h18"
                />
                <path
                  d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                />
                <path
                  d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                />
                <line
                  x1="10"
                  x2="10"
                  y1="11"
                  y2="17"
                />
                <line
                  x1="14"
                  x2="14"
                  y1="11"
                  y2="17"
                />
              </svg>
            </button>
          </div>
        </div>
      </li>
      <li>
        <div
          class="sc-eVqvcJ OmFqL"
        >
          <a
            aria-label="product-link"
            class="sc-bbbBoY kaFTmX"
            data-discover="true"
            href="/product/10"
            title="Centrifume Titanium Ring"
          >
            <div
              class="sc-cdmAjP kxojuH"
            >
              <img
                alt="Centrifume Titanium Ring"
                src="http://example.com/image-10.jpg"
              />
            </div>
            <div
              class="sc-jytpVa gpNucy"
            >
              <p
                class="sc-eknHtZ jjmWJx"
              >
                Centrifume Titanium Ring
              </p>
              <p
                class="sc-pYNGo jkBVsK"
              >
                $
                9.99
              </p>
            </div>
          </a>
          <div
            class="sc-cCVJLD cFLicH"
          >
            <button
              aria-label="cart-add-button"
              class="sc-jYczwO sc-dNFkOE bhIeZs jFefEh"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-shopping-cart"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="8"
                  cy="21"
                  r="1"
                />
                <circle
                  cx="19"
                  cy="21"
                  r="1"
                />
                <path
                  d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
                />
              </svg>
               Add to cart
            </button>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
